src/ray/core_worker/core_worker.h

```diff
@@ -684,6 +694,18 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
   /// be held up in garbage objects.
   std::function<void()> gc_collect_;
 
+  /// Callback to get the current language (e.g., Python) call site.
+  std::function<void(std::string *)> get_call_site_;
+
+  // Convenience method to get the current language call site.
+  std::string CurrentCallSite() {
```

Comment:
What's the reason for needing this callback into python? Can't we just pass through the callsite in the few places we know we're creating a new object (e.g., task submission, ray.put())? Seems like it would be a bit simpler and more general (not sure if this pattern would work in C++ for example)

Reply:
I think it creates more plumbing problems if you do it that way, vs. having a function you can call without needing to pass it through.